### PR TITLE
implemented recursive version of check_downstream

### PR DESCRIPTION
**Pull request by maximilianmordig**

### CODE REVIEW

**Comment by maximilianmordig** [comment]
This addresses issue #7

**Comment by waddella** [comment]
I have an issue when I select install all dependencies and then click the compute graph button:

image

**Comment by maximilianmordig** [comment]
There was a hard-coded dependency to this project dir. I fixed this.

**Review by waddella — COMMENTED** [review]
some comments above

File: R/dependencies.R [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+#' @md
+#' @inheritParams dependency_structure
+#' @inherit dependency_structure return
+#'
+#' @export
+#' @examples
+#' \dontrun{
+#' dependency_table()
+#' }
+dependency_table <- function(project = ".", feature = NULL,
```
[the remark this answers is not in the document]
Comment (maximilianmordig) [reply R/dependencies.R:635]: fixed

**Comment by maximilianmordig** [comment]
I did the following changes in the last commit:
replaced testthat::test_dir by devtools::test because they don't seem to agree (probably because namespace not loaded); fixed error when local_repos was NULL; added project to local_repos for shiny app; improved shiny app layout; run installation after shiny app closes as a job; compute dependency graph initially in the shiny app; put branch below repo label in graph; all addins now run as rstudio jobs rather than in the main session

**Comment by waddella** [comment]
thanks @maximilianmordig, I will test it and provide feedback

**Comment by waddella** [comment]
@maximilianmordig I will merge this PR and create new delta issues.

**Pull request merged**